ark/segmentation/marker_quantification.py

```diff
@@ -5,16 +5,156 @@
 
 import xarray as xr
 
-from skimage.measure import regionprops_table
+from skimage.measure import regionprops, regionprops_table
 
 from ark.utils import io_utils, load_utils, misc_utils, segmentation_utils
-from ark.segmentation.signal_extraction import extraction_function
+from ark.segmentation.signal_extraction import EXTRACTION_FUNCTION
+from ark.segmentation.regionprops_extraction import REGIONPROPS_FUNCTION
 
 import ark.settings as settings
 
 
+def compute_extra_props(props, regionprops_extras, **kwargs):
+    """Derives new features specified by regionprops_extras from a regionprops features
+
+    Args:
+        props (skimage.measure.regionprops):
+            A list of property information returned by regionprops
+        regionprops_extras (list):
+            A list of regionprops features to compute, each value should correspond to
+            a function in the global regionprops_functions variable
+        **kwargs:
+            Arbitrary keyword arguments
+
+    Returns:
+        pandas.DataFrame:
+            A dataframe with columns corresponding to each regionprops_extras feature
+    """
+
+    misc_utils.verify_in_list(
+        extras_props=regionprops_extras,
+        props_options=list(REGIONPROPS_FUNCTION.keys())
+    )
+
+    # define an empty list for each regionprop feature
+    prop_extra_data = {re: [] for re in regionprops_extras}
+
+    # generate the required data for each cell
+    for prop in props:
+        for re in regionprops_extras:
+            prop_extra_data[re].append(REGIONPROPS_FUNCTION[re](prop, **kwargs))
+
+    # convert the dictionary to a DataFrame
+    prop_extra_df = pd.DataFrame.from_dict(prop_extra_data)
+
+    return prop_extra_df
+
+
+def get_cell_props(segmentation_labels, regionprops_features, regionprops_extras, **kwargs):
+    """Gets regionprops features from the provided segmentation labels for a fov
+
+    Args:
+        segmentation_labels (numpy.ndarray):
+            rows x columns matrix of masks
+        regionprops_features (list):
+            morphology features for regionprops to extract for each cell
+        regionprops_extras (list):
+            list of extra properties derived from regionprops to compute
+        **kwargs:
+            Arbitrary keyword arguments for compute_extra_props
+
+    Returns:
+        pandas.DataFrame:
+            Contains the regionprops info (base and derived) for each labeled cell
+    """
+
+    cell_props = pd.DataFrame(regionprops_table(segmentation_labels,
+                                                properties=regionprops_features))
+
+    # compute the extras properties for each cell, and append to cell_props
+    props = regionprops(segmentation_labels)
+    extra_prop_data = compute_extra_props(props, regionprops_extras, **kwargs)
+    cell_props = pd.concat([cell_props, extra_prop_data], axis=1)
+
+    return cell_props
+
+
+def assign_cell_features(marker_counts, compartment, cell_props, cell_coords, cell_id,
```

Comment:
Same thing here, assign compartment features would be more clear. And potentially replacing all of the references to cell with compartment? Up to you

Reply:
I think it would be best to leave the cell references because those actually refer to a specific cell's id, coordinates, etc. However, the corresponding cell's data will be assigned to a specific compartment, which is specified as an argument to `assign_cell_features`.